### PR TITLE
feature-branch-backend into development

### PR DESCRIPTION
**Pull request by julianertle**
Backend for login and register ready to use

### CODE REVIEW

File: app/src/main/java/com/example/maraudersmap/UserController.kt [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+     */
+     suspend fun createNewUser(username : String, password : String, description : String): Response {
+
+        val userXTO = UserXTO()
+        userXTO.username = username
+        userXTO.password = password
+        userXTO.description = description
+
+        val server = ServerCommunicator()
+        val response = server.postRequest("https://maraudersmap-ext.hhn.dev/api/v0.2/user", userXTO)
```
Comment (FelixAlexK) [inline app/src/main/java/com/example/maraudersmap/UserController.kt:31]: replace url base with const and add the endpoints, like URL.plus("/user")

File: app/src/main/java/com/example/maraudersmap/LoginActivity.kt [hunk 2]
```diff
@@ -40,6 +52,43 @@ class LoginActivity : AppCompatActivity() {
...
 
+            val userController1 = UserController()
+            val response : Response = userController1.loginUser(username,password)      // sends a login request to the server and returns a response
+
+            val xmlBody = response.body!!.string()
+            val responseCode : Int = response.code      // Response codes: 200 = Login successful, 403 = Forbidden (Login failed), ? = Other unknown error codes possible
+
+            if(responseCode==200){
+                val userID: String = serializer.read(ExtractUserID::class.java, xmlBody).id.toString()
+                val jasonWebToken = response.headers.last().second
```
Comment (FelixAlexK) [inline app/src/main/java/com/example/maraudersmap/LoginActivity.kt:79]: typo in val name: jasonWebToken -> jsonWebToken

File: app/src/main/java/com/example/maraudersmap/UserXTO.kt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package com.example.maraudersmap
+
+internal class UserXTO {
```
Comment (FelixAlexK) [inline app/src/main/java/com/example/maraudersmap/UserXTO.kt:3]: Why internal class here ?

File: app/src/main/java/com/example/maraudersmap/UserController.kt [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+     * @param username username
+     * @param password password for the username
+     * @return Response of the request
+     */
+    suspend fun loginUser(username: String, password: String): Response {
+        val userXTO = UserXTO()
+        userXTO.username = username
+        userXTO.password = password
+
+        val server = ServerCommunicator()
```
Comment (FelixAlexK) [inline app/src/main/java/com/example/maraudersmap/UserController.kt:47]: Redundant declaration. Replace with global variable.

File: app/src/main/java/com/example/maraudersmap/RegisterActivity.kt [hunk 5]
```diff
@@ -45,6 +49,28 @@ class RegisterActivity : AppCompatActivity() {
...
+        registerUser("Username114","password","description")
 
     }
+
+    /**
+     * Sends a register request to the server and returns a response.
+     * The response code is saved in a variable
+     *
+     * @param username Username of the user
+     * @param password Password of the username
```
Comment (FelixAlexK) [inline app/src/main/java/com/example/maraudersmap/RegisterActivity.kt:61]: Missing parameter description

**Review by FelixAlexK — CHANGES_REQUESTED** [review]
Found some things, which have to be changed @julianertle

File: app/src/main/java/com/example/maraudersmap/UserController.kt (hunk 1 above)
Comment (julianertle) [reply app/src/main/java/com/example/maraudersmap/UserController.kt:32]: Needs to be dealt with later due to a bug

**Review by julianertle — COMMENTED** [review]
Resolved all but one comment due to a bug and made a new commit

**Pull request closed without merging**